File: moxy/compiler/src/main/java/com/omegar/mvp/compiler/entity/PresenterProviderMethod.java
package com.omegar.mvp.compiler.entity;

import com.omegar.mvp.presenter.PresenterType;

import javax.lang.model.type.DeclaredType;

public class PresenterProviderMethod {
	private final DeclaredType clazz;
	private final String name;
	private final PresenterType presenterType;
	private final String tag;
	private final String presenterId;

	public PresenterProviderMethod(DeclaredType clazz, String name, String type, String tag, String presenterId) {
		this.clazz = clazz;
		this.name = name;
		if (type == null) {
			presenterType = PresenterType.LOCAL;
		} else {
			presenterType = PresenterType.valueOf(type);
		}
		this.tag = tag;
		this.presenterId = presenterId;
	}

	public DeclaredType getClazz() {
		return clazz;
	}

	public String getName() {
		return name;
	}

	public PresenterType getPresenterType() {
		return presenterType;
	}

	public String getTag() {
		return tag;
	}

	public String getPresenterId() {
		return presenterId;
	}
}

File: moxy/compiler/src/main/java/com/omegar/mvp/compiler/entity/PresenterProviderMethod.kt
package com.omegar.mvp.compiler.entity

import com.omegar.mvp.compiler.extenions.orValueOf
import com.omegar.mvp.presenter.PresenterType
import javax.lang.model.type.DeclaredType

class PresenterProviderMethod(
    val clazz: DeclaredType,
    val name: String,
    val presenterType: PresenterType?,
    val tag: String?,
    val presenterId: String?
) {

    constructor(
        clazz: DeclaredType,
        name: String,
        type: String?,
        tag: String?,
        presenterId: String?
    ) : this(
        clazz,
        name,
        PresenterType.LOCAL.orValueOf(type),
        tag,
        presenterId
    )

}